tests/serverpod_test_server/test_e2e/auth_module_test.dart

```diff
@@ -172,4 +172,50 @@ void main() {
       expect(result, equals(true));
     });
   });
+
+  group('Given signed in user that wants to alter its user info', () {
+    setUp(() async {
+      var response = await client.authentication.authenticate(
+        'test@foo.bar',
+        'password',
+      );
+      assert(response.success, 'Failed to authenticate user');
+      await client.authenticationKeyManager
+          ?.put('${response.keyId}:${response.key}');
+      assert(
+          await client.modules.auth.status.isSignedIn(), 'Failed to sign in');
+    });
+
+    tearDown(() async {
+      await client.authenticationKeyManager?.remove();
+      await client.authentication.removeAllUsers();
+      await client.authentication.signOut();
+      assert(
+        await client.modules.auth.status.isSignedIn() == false,
+        'Still signed in after teardown',
+      );
+    });
+    test('when changing its user name it should succeed.', () async {
+      final String newUserName = 'newUserName';
+      var currentUserInfo = await client.modules.auth.status.getUserInfo();
+      var result = await client.modules.auth.user.changeUserName(newUserName);
+      var updatedUserInfo = await client.modules.auth.status.getUserInfo();
+      expect(result, equals(true));
+      expect(currentUserInfo?.userName, isNot(equals(newUserName)),
```

Comment:
Suggestion: Fetch the user name, modify it and then save it.

This test would break if we, for some probably stupid reason, modify the default user name to `newUserName`.
We can make it more robust by always retrieving whatever user name that already exists, update it with some suffix, and then fetch the user again to validate it is modified with our added suffix.

That makes the test independent from the outside environment.

Reply:
Same for other test.

Reply:
Makes a lot of sense, will do!

Reply:
Fixed in the second commit.